frontend/src/app/oauth/callback/page.tsx

```diff
@@ -1,12 +1,42 @@
 'use client'
 
 import {LitusOAuthCallback} from "@/utils/oauth";
-import {Suspense} from "react";
+import React, {Suspense, useEffect, useRef, useState} from "react";
+import {useRouter, useSearchParams} from "next/navigation";
+import ErrorPage from "@/components/error/ErrorPage";
 
 export default function OAuthCallbackPage() {
+    const router = useRouter();
+    const searchParams = useSearchParams();
+    const hasRun = useRef(false);
+    const [error, setError] = useState<Error>(null);
```

Comment:
error on null: TS2345: Argument of type null is not assignable to parameter of type Error | (() => Error)